pkg/storage/etcd/etcd_helper_test.go

```diff
@@ -87,271 +88,165 @@ func getEncodedPod(name string) string {
 	return string(pod)
 }
 
+func createObj(t *testing.T, helper etcdHelper, name string, obj, out runtime.Object, ttl uint64) error {
+	err := helper.Create(context.TODO(), name, obj, out, ttl)
+	if err != nil {
+		t.Errorf("Unexpected error %v", err)
+	}
+	return err
+}
+
+func createPodList(t *testing.T, helper etcdHelper, list *api.PodList) error {
+	for i := range list.Items {
+		returnedObj := &api.Pod{}
+		err := createObj(t, helper, list.Items[i].Name, &list.Items[i], returnedObj, 0)
+		if err != nil {
+			return err
+		}
+		list.Items[i] = *returnedObj
+	}
+	return nil
+}
+
 func TestList(t *testing.T) {
-	fakeClient := tools.NewFakeEtcdClient(t)
-	helper := newEtcdHelper(fakeClient, testapi.Default.Codec(), etcdtest.PathPrefix())
+	server := NewEtcdTestClientServer(t)
+	defer server.Terminate(t)
 	key := etcdtest.AddPrefix("/some/key")
-	fakeClient.Data[key] = tools.EtcdResponseWithError{
-		R: &etcd.Response{
-			EtcdIndex: 10,
-			Node: &etcd.Node{
-				Dir: true,
-				Nodes: []*etcd.Node{
-					{
-						Key:           "/foo",
-						Value:         getEncodedPod("foo"),
-						Dir:           false,
-						ModifiedIndex: 1,
-					},
-					{
-						Key:           "/bar",
-						Value:         getEncodedPod("bar"),
-						Dir:           false,
-						ModifiedIndex: 2,
-					},
-					{
-						Key:           "/baz",
-						Value:         getEncodedPod("baz"),
-						Dir:           false,
-						ModifiedIndex: 3,
-					},
-				},
-			},
-		},
-	}
-	expect := api.PodList{
-		ListMeta: unversioned.ListMeta{ResourceVersion: "10"},
+	helper := newEtcdHelper(server.client, testapi.Default.Codec(), key)
+
+	list := api.PodList{
 		Items: []api.Pod{
 			{
-				ObjectMeta: api.ObjectMeta{Name: "bar", ResourceVersion: "2"},
+				ObjectMeta: api.ObjectMeta{Name: "bar"},
 				Spec:       apitesting.DeepEqualSafePodSpec(),
 			},
 			{
-				ObjectMeta: api.ObjectMeta{Name: "baz", ResourceVersion: "3"},
+				ObjectMeta: api.ObjectMeta{Name: "baz"},
 				Spec:       apitesting.DeepEqualSafePodSpec(),
 			},
 			{
-				ObjectMeta: api.ObjectMeta{Name: "foo", ResourceVersion: "1"},
+				ObjectMeta: api.ObjectMeta{Name: "foo"},
 				Spec:       apitesting.DeepEqualSafePodSpec(),
 			},
 		},
 	}
 
+	createPodList(t, helper, &list)
 	var got api.PodList
-	err := helper.List(context.TODO(), "/some/key", storage.Everything, &got)
+	err := helper.List(context.TODO(), key, storage.Everything, &got)
 	if err != nil {
 		t.Errorf("Unexpected error %v", err)
 	}
-	if e, a := expect, got; !reflect.DeepEqual(e, a) {
+	if e, a := list.Items, got.Items; !reflect.DeepEqual(e, a) {
 		t.Errorf("Expected %#v, got %#v", e, a)
```

Comment:
IIUC, we are explicitly assuming here that etcd is returning objects in alphabetic order within a list. Can you please add a comment about since you're already touching this code?